docs/performance/proposal.md

```diff
@@ -0,0 +1,68 @@
+# Measuring Analyzer Performance
+
+Now that analyzers are part of the build we need a mechanism to track their performance across releases as well as build confidence regarding their use in the SDK.
+
+## Goals
+
+- Developers can quickly get feedback on how their change affects performance
+- We can track and detect performance regressions in builds before release.
+
+## What we do today
+
+- Roslyn
+  - Can be run on CI: **No**
+  - Can be run locally with a single script: **No**
+  - [Compiler Performance](https://github.com/dotnet/roslyn/blob/main/docs/wiki/Measuring-Compiler-Performance.md)
```

Comment:
Note that this benchmark also contains GetAnalyzerDiagnostics() scenario, which runs an old checked in FxCopAnalyzers package. I have primarily used that benchmark for all analyzer driver improvements, and also used it prior to .NET 5 release (I manually patched the dropped in analyzers and the benchmark to replace FxCopAnalyzers with NetAnalyzers assemblies). That benchmark did help significantly in tuning the performance for the .NET platform compat analyzer authored by @buyaa-n.